### ISSUE TITLE
Add Comments field to Costco Integration

### ISSUE THREAD

**Issue by kcaponeAP** [activeprospect/leadconduit-integration-costco#27]
Costco will be sending us information in the AM for 'comments'

**Comment by kcaponeAP** [comment]
Update! Costco requires 4 more fields.

Comments: 
    Type: varchar
    Example: 'This person needs a loan' - may also be blank.

Current_Mortgage 
    Type: string
    Allowed values:Yes, No
      
Initial_Draw
    Description: initial amount they want on closing
    Type: Integer [9]
    Example: 200000

Line_Amount 
   Description: total amount you want to be approved for
   Integer [11]
   Example: 10000000

**Comment by alexj** [comment]
@ros3bud How much work will this require?

**Comment by ros3bud** [comment]
Looks like it will require 1-2 hours tops.

**Comment by alexj** [comment]
Thanks! Can you knock it out this week, in between other projects?

**Comment by alexkwolfe** [comment]
Which LeadConduit fields should each of these be mapped to?

**Comment by kcaponeAP** [comment]
Initial_Draw --> mortgage.loan.cash_out
Comments --> comments

However, there are some other issues. These fields below are needed and there are no LCN fields for them. We'll need to have new LCN fields created to accommodate this request.

Current_Mortgage --> Should it be mortgage.current_mortgage
Line_Amount --> Should be mortgage.desired_loan_amount

**Comment by alexkwolfe** [comment]
How is `Line_Amount` different from `mortgage.loan.amount`?

**Comment by alexkwolfe** [comment]
How is `Current_Mortgage` different from `mortgage.first_mortgage_balance`?

**Comment by alexkwolfe** [comment]
Oh. `Current_Mortgage` just specifies if they have a mortgage, I guess.

**Comment by kcaponeAP** [comment]
The costco integration already uses mortgage loan amount.

**Comment by alexkwolfe** [comment]
> The costco integration already uses mortgage loan amount.
But what's the difference?

**Comment by alexkwolfe** [comment]
Under what circumstances is `Current_Mortgage` necessary? They already tell us `first_mortgage_balance`. 

I would think that when `first_mortgage_balance` is provided that `Current_Mortgage` will be "Yes" and when `first_mortgage_balance` is not provided that `Current_Mortgage` will be "No". If that's true, there's no case for adding a new field to capture `Current_Mortgage`. If that's not true, why not?

**Comment by kcaponeAP** [comment]
Line amount: the difference is that we need to separate out line amount from loan amount, since it is the line of credit desired. it is not a loan amount..whether its home improvement, etc. it's not considered a loan amount, since it may or may not be house related and in the grand scheme of things, I don't want this being routed to loan amount in velocify.

current mortgage: for costco's shorter forms, such as this one, they don't ask for mortgage balance. this is a yes/no question .. probably submitted from a mobile device. so they won't be sending first mortgage balance.

**Comment by alexkwolfe** [comment]
So `Current_Mortgage` and `first_mortgage_balance` won't be submitted together. OK.

**Comment by alexkwolfe** [comment]
Does Costco tell us which leads are short form?

**Comment by alexkwolfe** [comment]
How do other vendors provide the `Line_Amount` value?

**Comment by kcaponeAP** [comment]
Other vendors do not submit this field. However, we do have loan_amount, versus mortgage.loan_amount. would that work?

Costco tells us which leads are shortform, via the costco product flag:

shortform
mobile
finance
heloc
clivecall
and more to come..

when i say shortform, i am referring to any of these products. i just didn't want to add more confusion.

**Comment by alexkwolfe** [comment]
Is `Line_Amount` for a HELOC? Do other vendors really not submit HELOCs?

**Comment by kcaponeAP** [comment]
All of the requested fields for Costco are for their HELOC product. For HELOC, product=seek in the query string.

**Comment by ros3bud** [comment]
Can you please confirm the needed new fields and their mappings?

**Comment by kcaponeAP** [comment]
These are the ones that we have LCN fields for:

Initial_Draw --> mortgage.loan.cash_out
Comments --> comments

Comments: 
Type: varchar
Example: 'This person needs a loan' - may also be blank.

Initial_Draw
Description: initial amount they want on closing
Type: Integer [9]
Example: 200000

These ones require new LCN fields, as they don't match up with current available fields.  I do not know what the next step would be on incorporating this into LCN/Fields

Current_Mortgage --> Should it be mortgage.current_mortgage ?? **new field**
Line_Amount --> Should be mortgage.desired_loan_amount ?? **new field**

Current_Mortgage 
Type: string
Allowed values:Yes, No

Line_Amount 
Description: total amount you want to be approved for
Integer [11]
Example: 10000000

**Comment by alexkwolfe** [comment]
LendingTree provides home equity leads. That integration uses `mortgage.loan.amount`. How that different than what Costco is providing with their HELOC leads?

**Comment by kcaponeAP** [comment]
Im requesting information from Costco to find out if they ever are going to send line_amount and loan_amount at the same time and we'll go from there.

**Comment by kcaponeAP** [comment]
Costco replied with the following regarding line_amount:

"Our systems manager recommends you create different fields, but it is  not required. If you to map line_amount to loan _amount we can do that."

**Comment by alexkwolfe** [comment]
What question did you ask? If they ever send both fields at the same time?

**Comment by kcaponeAP** [comment]
Correct: "Would line_amount and loan_amount ever be used in the same lead?"
